src/sys.rs

```diff
@@ -431,14 +431,16 @@ sys_op! {
     (2(0), GifShow, Gifs, "&gifs", "gif - show", Mutating),
     /// Decode audio from a byte array
     ///
-    /// Returns the audio format as a string and an array representing the audio samples.
+    /// Returns the audio format as a string, the audio sample rate, and an array representing the audio samples.
     /// Only the `wav` format is supported.
     ///
     /// See also: [&ae]
-    (1(2), AudioDecode, Audio, "&ad", "audio - decode", Pure),
+    (1(3), AudioDecode, Audio, "&ad", "audio - decode", Pure),
     /// Encode audio into a byte array
     ///
-    /// The first argument is the format, and the second is the audio samples.
+    /// The first argument is the format, the second is the audio sample rate, and the third is the audio samples.
+    ///
+    /// The sample rate must be a 32-bit positive integer.
```

Comment:
I don't think the size of the integer should be specified. Nothing else in the docs defines constraints in terms of the size of integers.
I'd prefer just saying "must be a positive integer".
Then, if it is too high to fit in the u32, just error with something like "sample rate is too high".